### PR TITLE
Clean build warnings

### PR DESCRIPTION
**Pull request by cydrain**
Issue: #443

### CODE REVIEW

File: thirdparty/hnswlib/hnswlib/hnswalg.h [hunk 1]
```diff
@@ -1392,7 +1392,7 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
         }
 
         std::unique_ptr<int8_t[]> query_data_sq;
-        const data_t* raw_data = (const data_t*)query_data;
+        [[gnu::unused]] const data_t* raw_data = (const data_t*)query_data;
```
Comment (PwzXxm) [inline thirdparty/hnswlib/hnswlib/hnswalg.h:1395]: [[maybe_unused]]? Why use gnu directives? Clang may fail

Comment (cydrain) [reply thirdparty/hnswlib/hnswlib/hnswalg.h:1395]: change to [[maybe_unused]]

**Pull request merged**